google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -161,6 +160,7 @@ def __init__(
         self,
         instance_connection_string: str,
         driver_name: str,
+        keys: concurrent.futures.Future,
```

Comment:
`concurrent.futures.Future` isn't Awaitable, so you'll need to make a coro with `loop.call_soon_threadsafe`) and only use it from asyncio-land.

Reply:
I'm not sure how to get `call_soon_threadsafe` to work for this. It returns an `asyncio.Handle` which isn't Awaitable.  Any reason why awaiting the wrapped future from the asyncio thread won't work?

Reply:
Ah my bad --  I didn't realize this was in the `__init__` signature. So the two options are:

1. Use a future and wrap it as a Task. This makes the future more async compatible but also introduces more overhead (since futures are thread level).

2. Pass in a coro directly and start it with thread_safe. This is less overhead but blurs the lines between thread-land and asyncio-land. 

Not sure if there is a clear winner.

Reply:
I think this is probably fine.